### PR TITLE
spidershim: implement v8::Value::IsFunction()

### PR DESCRIPTION
**Pull request by tbsaunde**

### CODE REVIEW

File: deps/spidershim/src/v8value.cc [hunk 1]
```diff
@@ -323,4 +323,12 @@ bool Value::SameValue(Handle<Value> that) const {
   return same;
 }
 
+bool
+Value::IsFunction() const
+{
+  JSContext* cx = JSContextFromIsolate(Isolate::GetCurrent());
+  const JS::RootedValue thisVal(cx, *reinterpret_cast<const JS::Value*>(this));
+  return thisVal.isObject() && JS::IsCallable(&thisVal.toObject());
```
Comment (ehsan) [inline deps/spidershim/src/v8value.cc:331]: Since `IsCallable` doesn't need the handle, you don't need to root `thisVal` like you have done here.

**Comment by ehsan** [comment]
Sorry, I already implemented this last night in b8035918496541ef436d3c29e2ff28432f56ee7a.

**Comment by ehsan** [comment]
Also please add tests alongside every function/class being implemented.  Thanks!

**Comment by Ryuno-Ki** [comment]
Sorry for commenting on an already closed PR, but to prevent further occurences of https://github.com/mozilla/spidernode/pull/38#issuecomment-213470949 wouldn't it be prudent to add "Good first bug" or "Help welcome" labels to issues the Mozilla team does not intend to implement on their own (quickly)?

This way, contributors wouldn't be driven away or waste their time writing code for what already gets addressed.

**Comment by ehsan** [comment]
> Sorry for commenting on an already closed PR, but to prevent further occurences of #38 (comment) wouldn't it be prudent to add "Good first bug" or "Help welcome" labels to issues the Mozilla team does not intend to implement on their own (quickly)?
> 
> This way, contributors wouldn't be driven away or waste their time writing code for what already gets addressed.

Yeah, good point.  So far most of the issues I have filed are ones that Mozilla is interested in implementing pretty quickly, and given that a lot of the V8 embedding APIs depend on each other, it's hard to coordinate efforts so that people don't step on each other's toes, but as we file more issues that are not things we immediately need to do, I'll try to mark them as "help-wanted" to make it clear what the community can pick up.  I should also go through the existing list of issues and separate out such parts for the already filed ones.  I'm traveling this week so it may be tough to do that before next week.

In the mean time if you're interested in working on something please feel free to reach out to me, and I'll try to help to make sure nobody else is working on the same thing.  :)

**Comment by Ryuno-Ki** [comment]
> In the mean time if you're interested in working on something please feel free to reach out to me

No, I stay as far away from C++ as possible. I'm awaiting a running _Hello world_ app and look into porting things (I'm rather familiar with JavaScript).

**Pull request closed without merging**